### ISSUE TITLE
Supervised graph classification with GCN produce different output for the same input

### ISSUE THREAD

**Issue by evan42mr** [stellargraph/stellargraph#1783]
### Describe the bug
In the notebook example the train and test data are created by RepeatedStratifiedKFold. After that it runs in a training loop for repeated k-fold cross-validation. The final result is given as accuracy over all folds mean and std. 

I ran training loop for repeated k-fold cross-validation with the **same train and test data** 50 times. As the result I expected that the same accuracy for each training, however some of the results were different with deviation of 6%.

Is it possible to secure that the model output will always be the same for the same train and test input data set?

### To Reproduce

1. *Go to Supervised graph classification with GCN on the website*
2. *Open notebook in colab*
3. *Instead of creating new train and test data with every training loop iteration use the same train and test data.*

**Comment by huonw** [comment]
Thanks for filing an issue!

There's several sources of a randomness in a deep learning model, with train/test split being just one. For instance, the initial values of the trainable parameters. Most sources of randomness can be controlled via `tensorflow.set_seed` and [`stellargraph.random.set_seed`](https://stellargraph.readthedocs.io/en/stable/api.html#stellargraph.random.set_seed). They should allow making the model outputs far more reproducible.

**Comment by evan42mr** [comment]
Yes, the outputs became more reproducible. Thank you
